src/interrupt/legacy.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2021 Amazon.com, Inc. or its affiliates.
+// All Rights Reserved.
+
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Traits and Structs to manage legacy interrupt sources for devices.
+//!
+//! Legacy interrupt sources typically include pin based interrupt lines.
+
+use crate::interrupt::ConfigurableInterrupt;
+
+/// Definition for PCI INTx pins.
+#[derive(Copy, Clone, Debug)]
+pub enum IntXPin {
+    /// INTA
+    INTA = 0x1,
+    /// INTB
+    INTB = 0x2,
+    /// INTC
+    INTC = 0x3,
+    /// INTD
+    INTD = 0x4,
+}
+
+/// Standard configuration for Legacy interrupts.
+#[derive(Copy, Clone, Debug, Default)]
+pub struct LegacyIrqConfig {
+    /// Input of the system interrupt controllers the device's interrupt pin is connected to.
+    /// Implemented by any device that makes use of an interrupt pin.
+    pub interrupt_line: Option<u32>,
+    /// Specifies which interrupt pin the device uses.
+    pub interrupt_pin: Option<IntXPin>,
+}
+
+/// Supertrait for defining properties of Legacy interrupts.
```

Comment:
`LegacyInterrupt ` is not a supertrait, it's a subtrait. The readme correctly states that `ConfigurableInterrupt` and `MaskableInterrupt` are supertraits. I would not focus on this here and just call it a trait.

Reply:
Yup, you're right. Best to leave it as "Trait".

Reply:
Corrected.